File: baekjoon/다이나믹_프로그래밍/2565-전깃줄.py
wire_count = int(input())
connect = [0 for _ in range(501)]

for _ in range(wire_count):
    a, b = map(int, input().split())
    connect[a] = b

sequence = [conn for conn in connect if conn != 0]

def lis():
    dp = [1 for _ in range(wire_count)]

    for i in range(wire_count):
        for j in range(0, i):
            if sequence[j] < sequence[i]:
                dp[i] = max(dp[i], dp[j] + 1)
    
    return max(dp)

print(wire_count - lis())

File: baekjoon/다이나믹_프로그래밍/2565-전깃줄.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    private static int wireCount;
    private static List<Integer> sequence = new ArrayList<>();

    public static void main(String[] args) {
        init();
        System.out.println(wireCount - lis());
    }

    private static void init() {
        Scanner sc = new Scanner(System.in);
        wireCount = sc.nextInt();
        int[] connect = new int[501];

        for (int i = 0; i < wireCount; i++) {
            int a = sc.nextInt();
            int b = sc.nextInt();

            connect[a] = b;
        }

        for (int dest : connect)
            if (dest != 0)
                sequence.add(dest);
    }

    private static int lis() {
        int[] dp = new int[wireCount];

        int lisCount = 1;
        for (int i = 0; i < wireCount; i++) {
            dp[i] = 1;
            for (int j = 0; j < i; j++) {
                if (sequence.get(j) < sequence.get(i)) {
                    dp[i] = Math.max(dp[i], dp[j] + 1);
                    lisCount = Math.max(lisCount, dp[i]);
                }
            }
        }

        return lisCount;
    }
}